src/bucket.js

```diff
@@ -2326,15 +2327,32 @@ class Bucket extends common.ServiceObject {
     }
 
     function upload() {
-      let sourceStream;
-
       if (isURL) {
-        sourceStream = request.get(requestOptions);
-      } else {
-        sourceStream = fs.createReadStream(pathString);
+        const url = requestOptions.url;
+        const method = 'GET';
+        const headers = requestOptions.headers;
```

Comment:
This feels like a lot of boilerplate.  Instead of extracting url, method, and headers from the request, why not just pass the requestOptions directly?

Reply:
Being verbose to show which options are understood by `fetch`. Except for the most basic ones they have different names.

Reply:
I would vote for being simple here. If it's possible to just pass the user's options directly through to node-fetch, that would be the best for all of us:

- Easy docs (just link to node-fetch's docs)
- Simple code (no mapping)